internal/transport/http2_client.go

```diff
@@ -1006,24 +1008,34 @@ func (t *http2Client) Close(err error) {
 		t.kpDormancyCond.Signal()
 	}
 	t.mu.Unlock()
+
+	var st *status.Status
 	// Per HTTP/2 spec, a GOAWAY frame must be sent before closing the
-	// connection. See https://httpwg.org/specs/rfc7540.html#GOAWAY.
+	// connection. See https://httpwg.org/specs/rfc7540.html#GOAWAY. It
+	// also waits for loopyWriter to be closed with a timer to avoid the
+	// long blocking in case the connection is half closed.
```

Comment:
"in case the connection is half closed" is incorrect, right?  Isn't it needed in case the connection is blackholed, i.e. TCP is just stuck?

Reply:
Updated